src/util/locking/RedisResourceLocker.ts

```diff
@@ -0,0 +1,158 @@
+import { assert } from 'console';
+import type { RedisClient } from 'redis';
+import { createClient } from 'redis';
+import type { Lock } from 'redlock';
+import Redlock from 'redlock';
+import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
+import { getLoggerFor } from '../../logging/LogUtil';
+import { InternalServerError } from '../errors/InternalServerError';
+import type { ResourceLocker } from './ResourceLocker';
+
+// The ttl set on a lock
+const ttl = 1000;
```

Comment:
Locks are extended indefinitely, so this shouldn't matter.
Settings this to eg: 100sec, even if a lock was released immediately, the mechanism to extend the lock indefinitely wil keep a promise open for 50sec.

Reply:
You're right, that would indeed be a problem. So I propose that the timeouts get stored with together with the lock in the locklist. That way you can use `clearTimeout` when the lock gets released. That way we can use a higher value (so less resets have to happen) while not having a bunch of timeouts run when not necessary.

Reply:
I found that because of the async() function in the setTimeout(), we can not cancel/abort it with the clearTimeout() function.

I think we better leave the ttl at 2s because most operations wont take that long.